integration/users/connection_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package users
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+)
+
+func TestAuthentication(t *testing.T) {
+	t.Parallel()
+
+	s := setup.SetupWithOpts(t, &setup.SetupOpts{})
+	ctx := s.Ctx
+	collection := s.Collection
+	db := collection.Database()
+
+	testCases := map[string]struct { //nolint:vet // for readability
+		username       string
+		password       string
+		updatePassword string // if true, the password will be updated to this one after the user is created.
+		mechanism      string
+
+		wrongPassword bool
+	}{
+		"Success": {
+			username: "common",
+			password: "password",
+		},
+		"Updated": {
+			username:       "updated",
+			password:       "pass123",
+			updatePassword: "somethingelse",
+		},
+		"Scramsha256": {
+			username:  "scramsha256",
+			password:  "password",
+			mechanism: "SCRAM-SHA-256",
+		},
+		"Scramsha256updated": {
+			username:       "scramsha256updated",
+			password:       "pass123",
+			updatePassword: "anotherpassword",
+			mechanism:      "SCRAM-SHA-256",
+		},
+		"BadPassword": {
+			username:      "baduser",
+			password:      "something",
+			mechanism:     "SCRAM-SHA-256",
+			wrongPassword: true,
```

Comment:
Do we test a wrong password for an existing user too?

Reply:
Yes. This tested user actually exists.